cfn-resources/project/cmd/resource/resource.go

```diff
@@ -264,55 +264,45 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (event h
 	}
 
 	if currentModel.ProjectApiKeys != nil {
-		// Get APIKeys from project
-		itemsPerPage := 1000
-		projectAPIKeys, _, err := atlasV2.ProgrammaticAPIKeysApi.ListProjectApiKeysWithParams(context.Background(), &admin.ListProjectApiKeysApiParams{
-			GroupId:      projectID,
-			ItemsPerPage: &itemsPerPage,
-		}).Execute()
-		if err != nil {
-			_, _ = logger.Warnf("ProjectId : %s, Error: %s", projectID, err)
-			return handler.ProgressEvent{
-				OperationStatus:  handler.Failed,
-				Message:          "Error while finding api keys in project",
-				HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
-		}
-
 		// Get Change in ApiKeys
-		newAPIKeys, changedKeys, removeKeys := getChangeInAPIKeys(*currentModel.Id, currentModel.ProjectApiKeys, projectAPIKeys.Results)
+		newAPIKeys, changedKeys, removeKeys := getChangeInAPIKeys(currentModel.ProjectApiKeys, prevModel.ProjectApiKeys)
```

Comment:
Only concern I have is that how configuration drift would be handled. If for example an api key is created from the Atlas UI for this project, my understanding is this stack would never contemplate this new key right? Because it does not fetch all api keys.
Taking this drawback into account, I still believe this new behaviour is better that having the default project owner key deleted on any update.